### ISSUE TITLE
`gitui` crashes with too many untracked files

### ISSUE THREAD

**Issue by glaeqen** [extrawurst/gitui#1542]
**Describe the bug**
This problem becomes an issue when attempting to use `gitui` together with tools like `vcsh`. VCSH is a git wrapper that uses "fake" bare git repositories that set a root in a home directory. Thus, one can modularize system configuration by having multiple repositories with a root at $HOME.

Trying to run `gitui` through `vcsh run <repo_name> gitui` (this sets GIT_DIR accordingly) results in a crash because it tries to render all untracked files in the home directory.

**To Reproduce**
Steps to reproduce the behavior (without vcsh):
1. Run
```sh
DIR="/tmp/some-test-bare-repo.git"
git init --bare "$DIR" && \
GIT_DIR="$DIR" GIT_WORK_TREE="$HOME" gitui
```
Note: $HOME has to be big enough to cause the crash. Otherwise, any other directory with huge amount of files pointed to by `$GIT_WORK_TREE` could be used.

2. Crash

**Expected behavior**
Crash caused by immense amount of files is probably very difficult to solve. Nonetheless some kind of workaround would be nice. For example, similarly to `git status --untracked-files=no`, a new flag for `gitui` could make it ignore all the untracked files. Feel free to suggest the solution though :)

**Comment by glaeqen** [comment]
I initially commented at #1223, FTR

**Comment by glaeqen** [comment]
I just realized that even with `.gitignore.d/<repo>` set to ignore everything *but* checked in files, `gitui` crashes with

```
Error: receiving on an empty and disconnected channel
```

**Comment by extrawurst** [comment]
can you try running with `--polling`?
see https://github.com/extrawurst/gitui/issues/1444

**Comment by glaeqen** [comment]
Yey, it helped! Although running `vcsh write-gitignore <repo>` was necessary to make irrelevant part of the $HOME dir invisible for `git` . Otherwise `gitui` starts choking on files. But does not crash! Thanks

**Comment by extrawurst** [comment]
How exactly does it behave when you say “choking on files”? Even with the polling arg?

**Comment by glaeqen** [comment]
Well, working directory is in $HOME. So it takes a minute or so to even list the files that are untracked. 
![image](https://user-images.githubusercontent.com/80407822/220174882-c12a14ce-4fbf-4ae0-9f8a-c61622da4fdd.png)

If all untracked files are ignored, it is pretty much instantaneous.

**Comment by extrawurst** [comment]
should be fixed in the next release via #1444
